src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectWriter.cs

```diff
@@ -78,15 +73,9 @@ internal void Serialize(object graph, Header[] inHeaders, BinaryFormatterWriter
             _serObjectInfoInit = new SerObjectInfoInit();
 
             _topId = InternalGetId(graph, false, null, out isNew);
-            headerId = _headers != null ? InternalGetId(_headers, false, null, out isNew) : -1;
+            headerId = -1;
```

Comment:
(Seems like the headerId we're now passing around won't actually be used anywhere, but to avoid churning a lot of code that we'd like to keep relatively consistent with desktop, it probably makes sense to stop where you did.)